src/upload.ts

```diff
@@ -254,32 +279,46 @@ export async function uploadLargeFileRequest(
   // the part-split function. Note that each part has to be chunk-aligned, so we
   // may limit the number of parallel uploads.
   let parallelUploads = 1;
+  let staggerPercent: number | null = null;
```

Comment:
`nit` don't we want to default to something here?

Reply:
50% would probably be a safe value for most connection speeds.